### DISCUSSION TITLE
Week 6 Report

### DISCUSSION THREAD

**Discussion by derrrobin** [SE-TINF22B6/CookPit#33] [General]
`This blog will be updated in the week.`

We have used the following link in our diagram:

- Dependency
- Association
- Composition
- Aggregation

Between the class `Upload` and `User`, we have a composition because the upload of a recipe depends on a user. Without a user account, it is not possible to upload a recipe.

In contrast, we have an aggregation between the `Recipe` and `Ingredient` classes. At first, you might think that an ingredient does not exist without a recipe. However, this is not the case, as an ingredient does not necessarily depend on the existence of a recipe.

We have created a dependency link between the `Role` and `Upload` classes. This is because the possibility of uploading a recipe depends on the role, which uses the role of a user to decide whether an upload is possible or not.

Also we have inserted a normal association between the other roles.

`This week we also uploaded a better version of our OpenAI sequence diagrams.`

Initially, we had only linked the website to the API. Now we have added a link to the user describing how the user interacts with the website and how the website interacts with the API.

You can find the improved version here : https://github.com/SE-TINF22B6/CookPit/blob/main/Docs/SRS%20Images/sequence-diagram-api.png

# Unser Fortschritt diese Woche

This week, we've made significant progress in implementing the first components of the front end. Here's a summary of what we've achieved:

- **Header Addition:**
  - A header has been successfully added to the front end. The header includes:
    - Menu
    - Logo/Website Name
    - Language Selection
    - Favorites Button
    - Login Button

- **Routing Preparations:**
  - Initial preparations have been made for the upcoming routing requirements. This lays the foundation for smooth navigation within the website.

- **Global Project Presentation Readiness:**
  - To showcase our project on a global scale in the future, we've established a server. This server will host the website, ensuring its availability for the upcoming presentation.

**Database Design**
Moreover, we have made sginificant progress in the development of our PostgreSQL database. We have uploaded to this repository the relational diagram of the different tables. In addition, we have written the starting SQL scripting to generate the tables in the database. Furthermore, we have written some basic stored procedures to start making queries in the database. You can check the changes in the following link:

https://github.com/SE-TINF22B6/CookPit/tree/main/server/Database

We're excited about the progress and look forward to presenting our work with the new features and functionalities. The server is set up and ready to go, ensuring a seamless experience for our users during the presentation.

**Comment by BenjaminB109** [comment]
Dear CookPit Team,
sounds great. Make sure you have everything in English in the weekly report, so you should translate "Unser Fortschritt diese Woche" next time. Keep it up!

**Comment by Stev3eM** [comment]
Hello CookPitters,

that was a very extended blog post this week. Well done regarding to the level of information given but you should try to stick with one marketing / documentation language (_big headline: 'Unser Fortschritt diese Woche'_). This could turn a very informative and good written post into an amateur one, perhaps. 

How ever I'm glad to have seen the first frontend elements implemented into the project. A little impression would have been nice either also it could hook new stakeholders to keep with this blog further more.

**Comment by fehlend0** [comment]
Dear CookPit Team,

We find that the explanation of dependency, association, composition, and aggregation between classes, as well as the examples are pretty clear. Moreover, you’ve made great progress in the frontend. Providing a relational diagram of different tables, SQL scripting to generate tables, and basic stored procedures for queries is a great foundation for your future database development. All in all, we find your project well-rounded and well-documented.

 We look forward to seeing your future progress!

Your Spotify Team
